server/src/main.rs

```diff
@@ -44,6 +44,22 @@ async fn main() {
     ))
     .expect("could not create db connection");
 
+    let authorized = with_db(database.clone())
+        .and(warp::cookie("token"))
+        .and_then(|db: db::Pool, token: String| async move {
+            let res = data::author::dsl::author
+                .filter(data::author::dsl::access_token.eq(&token))
+                .execute(&db.get().unwrap());
```

Comment:
I reckon you can do this with `first`
```suggestion
                .select(data::author::id)
                .first::<i32>(&db.get().unwrap());
```

Reply:
What is returned here if the `user` doesn't exist?

Reply:
It's an error.

Reply:
https://docs.rs/diesel/1.4.5/diesel/result/enum.Error.html#variant.NotFound